__tests__/feature/chat-streaming.spec.jsx

```diff
@@ -148,42 +220,65 @@ describe('AppBase, streaming events execute properly', () => {
   test(`message receival through streamStart streamData streamEnd work for message with [${intent.type.payment}]`, () => {
     // Act
     act(() => {
-      localStorage.setItem("__pd", JSON.stringify({
-        "amount": "109.00",
-        "amountInUSD": 109,
-        "currency": "USD",
-        "billingOptionType": "one-time",
-        "provider": "primer",
-        "planId": "22641f3b-1474-4c1b-9c90-288f3f572f00",
-        "displayPlanPrice": "$109"
-      }))
-      root = renderWithProviders(<div id="chatbot-container"><AppBase config={getInitialConfig({ id: uuidV4(), purpose: '', close: { visible: true } })} /></div>);
-      root.store.dispatch(setConnected(true));
-      jest.advanceTimersByTime(STORING_CHECKER_INTERVAL * 5)
+      localStorage.setItem(
+        '__pd',
+        JSON.stringify({
+          amount: '109.00',
+          amountInUSD: 109,
+          currency: 'USD',
+          billingOptionType: 'one-time',
+          provider: 'primer',
+          planId: '22641f3b-1474-4c1b-9c90-288f3f572f00',
+          displayPlanPrice: '$109',
+        }),
+      );
+      root = renderWithProviders(
+        <div id="chatbot-container">
+          <AppBase config={getInitialConfig({ id: uuidV4(), purpose: '', close: { visible: true } })} />
+        </div>,
+      );
+
+      jest.advanceTimersByTime(STORING_CHECKER_INTERVAL * 5);
     });
 
     act(() => {
       serverSocket.emit(Events.chatHistory, {
-        "region": faker.location.country(),
-        "history": [],
-        "errors": []
-      })
-      jest.advanceTimersByTime(((initialMessage.length * 1000) * initialMessage.length - 1));
+        region: faker.location.country(),
+        history: [],
+        errors: [],
+        term: term,
+        threadId: threadId,
+      });
+      jest.advanceTimersByTime(initialMessage.length * 1000 * initialMessage.length - 1);
     });
 
     act(() => {
-      dispatchStreaming([{
-        "type": "text",
-        "text": "",
-        "sequence": 1,
-        "id": "4216d0a8-2621-499c-8a65-70191e31ee5a"
-      }, {
-        "type": "text",
-        "text": " ticket",
-        "sequence": 1,
-        "id": "4216d0a8-2621-499c-8a65-70191e31ee5a"
-      },
-        { "id": "4216d0a8-2621-499c-8a65-70191e31ee5a", 'type': 'payment', 'sequence': 1, 'payment': 'provide payment' }]);
+      dispatchStreaming([
+        {
+          type: 'text',
+          text: '',
+          sequence: 1,
+          id: '4216d0a8-2621-499c-8a65-70191e31ee5a',
+          term,
+          threadId: threadId,
+        },
+        {
+          type: 'text',
+          text: ' ticket',
+          sequence: 1,
+          id: '4216d0a8-2621-499c-8a65-70191e31ee5a',
+          term,
+          threadId: threadId,
+        },
+        {
+          id: '4216d0a8-2621-499c-8a65-70191e31ee5a',
+          type: 'payment',
+          sequence: 1,
+          payment: 'provide payment',
+          term,
+          threadId: threadId,
+        },
+      ]);
     });
 
```

Comment:
@IlievaDayana I see a lot of duplicate objects. You can create factory function that produces mocked objects and call it wherever needed:

const generateStreamingData = (data: Partial<StreamingData>) => ({
          type: 'text',
          text: '',
          sequence: 1,
          id: '4216d0a8-2621-499c-8a65-70191e31ee5a',
          term: '',
          threadId: faker.random.uuid(),
          ...data
});